includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1482,16 +1482,42 @@ public function set_payment_method_title_for_order( $order, $payment_method_type
 	 * @return array Array of keyed metadata values.
 	 */
 	protected function get_metadata_from_order( $order, $payment_type ) {
+		switch ( static::class ) {
+			case UPE_Split_Payment_Gateway::class:
+				$gateway_type = 'split_upe';
+				break;
+			case UPE_Payment_Gateway::class:
+				$gateway_type = 'upe';
+				break;
+			default:
+				$gateway_type = 'classic';
+		}
+
+		switch ( $order->get_created_via() ) {
```

Comment:
Is this `switch` statement really needed? Sending `checkout` and `store-api` to the server doesn't seem bad by itself. Also, even if something else happens, we'd never see `unknown`.